packages/storybook-framework-qwik/src/docs/qwik-docgen.ts

```diff
@@ -0,0 +1,37 @@
+import { parse } from "react-docgen-typescript";
+import type { PluginOption } from 'vite';
+import { createFilter } from 'vite';
+import MagicString from 'magic-string';
+
+export function qwikDocgen(): PluginOption {
+    // Naive way to include files containing Qwik component (by excluding stories).
+    // Will not work with files named 'story' or something else.
+    const isComponent = /^(?!.*\.stories\.tsx$).*\.tsx$/;
```

Comment:
As the comment acknowledges, this is a little brittle. The story file pattern can be configured, so I'd rather avoid making assumptions about it. (although `stories.tsx` is definitely the most common pattern).

Maybe we could just match all tsx files? Would there be a downside to that?

Reply:
Fixed by passing stories declared in `defaultConfig.optimizeDeps.entries` and excluding them within `qwikDocgen `.

Reply:
Good solution!